runner/ollamarunner/runner.go

```diff
@@ -1208,9 +1208,15 @@ func (s *Server) allocModel(
 		return err
 	}
 
-	if !s.cache.enabled && parallel > 1 {
-		parallel = 1
-		slog.Warn("model does not support caching, disabling parallel processing")
+	if !s.cache.enabled {
+		if parallel > 1 {
+			parallel = 1
+			slog.Warn("model does not support caching, disabling parallel processing")
+		}
+		if s.batchSize < kvSize {
+			s.batchSize = kvSize
```

Comment:
We actually use both parallel and batchSize in the NewInputCache function, so we should not change their values after it has already been initialized. To avoid this, we should probably move this whole `if !s.cache.enabled` block before NewInputCache and check `model.Config().Cache != nil` as the condition instead.